src/main/java/org/springframework/data/redis/cache/DefaultRedisCacheWriter.java

```diff
@@ -104,7 +105,11 @@ public void put(String name, byte[] key, byte[] value, @Nullable Duration ttl) {
 			if (shouldExpireWithin(ttl)) {
 				connection.set(key, value, Expiration.from(ttl.toMillis(), TimeUnit.MILLISECONDS), SetOption.upsert());
 			} else {
-				connection.set(key, value);
+				if (shouldExpireMaxIdleWithin(maxIdle)) {
+					connection.set(key, value, Expiration.from(maxIdle), SetOption.upsert());
+				} else {
+					connection.set(key, value);
+				}
```

Comment:
Expiration (and specifically, TTL) is something that is already [handled](https://github.com/spring-projects/spring-data-redis/blob/3.0.8/src/main/java/org/springframework/data/redis/cache/DefaultRedisCacheWriter.java#L104-L105) as of Spring Data Redis `2.7.x` (and beyond: `3.0.x`, `3.1.x`, and currently, `3.2.x`), as is apparent from line 105-107, above.